core/state_transition.go

```diff
@@ -954,21 +954,22 @@ func (st *stateTransition) innerExecute() (*ExecutionResult, error) {
 
 			amtU256 = st.collectNativeBalance(amtU256)
 			st.state.AddBalance(params.OptimismBaseFeeRecipient, amtU256, tracing.BalanceIncreaseRewardTransactionFee)
-			if l1Cost := st.evm.Context.L1CostFunc(st.msg.RollupCostData, st.evm.Context.Time); l1Cost != nil {
-				amtU256, overflow = uint256.FromBig(l1Cost)
-				if overflow {
-					return nil, fmt.Errorf("optimism l1 cost overflows U256: %d", l1Cost)
-				}
-				if shouldCheckGasFormula {
+			if shouldCheckGasFormula {
```

Comment:
I guess you can put all `shouldCheckGasFormula` checks in one condition check rather than multiple ones.